api/src/main/java/org/openmrs/api/db/hibernate/HibernateContextDAO.java

```diff
@@ -93,126 +93,135 @@ public void setUserDAO(UserDAO userDao) {
 	@Override
 	@Transactional(noRollbackFor = ContextAuthenticationException.class)
 	public User authenticate(String login, String password) throws ContextAuthenticationException {
-		String errorMsg = "Invalid username and/or password: " + login;
+		if (StringUtils.isBlank(login) || password == null) {
```

Comment:
I would not do this validation as part of this ticket. It would be safer to limit its scope to the ticket requirements. Any other improvements of the code base can have separate tickets and pull requests. 
Did you get a chance to look at this? https://opensource.com/article/18/6/anatomy-perfect-pull-request